pkg/allocator/allocator.go

```diff
@@ -389,10 +389,6 @@ func (a *Allocator) GetEvents() AllocatorEventChan {
 func (a *Allocator) Delete() {
 	close(a.stopGC)
 	a.mainCache.stop()
-
-	if a.events != nil {
-		close(a.events)
-	}
```

Comment:
This should've been cleaned up on L266 anyway in the Close() call so why move it outside of the `Delete()` function.

Reply:
np,  so imho the right thing to do when closing the remoteCache would be to delete the whole remoteAllocator - i.e. running `remoteAlloc.Delete()` -- but the problem with that is that without moving this code outside it would close the event channel not owned by the remoteAlloc, but owned by the main `CachingIdentityAllocator`. This in fact breaks the event channel which is used to pass remoteIdentity events to the main allocator (and panics on writing to closed channel).